APKBUILD

```diff
@@ -3,19 +3,22 @@
 pkgname="glibc"
 pkgver="2.35"
 _pkgrel="0"
-pkgrel="0"
+pkgrel="1"
 pkgdesc="GNU C Library compatibility layer"
+options="lib64"
 arch="x86_64"
 url="https://github.com/sgerrand/alpine-pkg-glibc"
 license="LGPL"
 source="https://github.com/sgerrand/docker-glibc-builder/releases/download/$pkgver-$_pkgrel/glibc-bin-$pkgver-$_pkgrel-x86_64.tar.gz
 nsswitch.conf
 ld.so.conf"
-subpackages="$pkgname-bin $pkgname-dev $pkgname-i18n"
+subpackages="$pkgname-bin $pkgname-utils $pkgname-dev $pkgname-i18n"
 triggers="$pkgname-bin.trigger=/lib:/usr/lib:/usr/glibc-compat/lib"
 
 package() {
-  mkdir -p "$pkgdir/lib" "$pkgdir/usr/glibc-compat/lib/locale"  "$pkgdir"/usr/glibc-compat/lib64 "$pkgdir"/etc
+  provides="libc6-compat"
```

Comment:
This declaration isn't accurate but the line below it is.
```suggestion
```

Reply:
I'm not sure I understand why you think this is incorrect.

In the PR description I wrote:

> the intent of `libc6-compat` can be written as "I'd like to be able to run binaries that are linked against glibc", so also say `provides="libc6-compat"`.

Or, put another way: What could be more libc6-compatible than actually installing libc6?

Reply:
> What could be more libc6-compatible than actually installing libc6?

`provides` is not meant to declare a capability, but should refer to packages provided, see also https://wiki.alpinelinux.org/wiki/APKBUILD_Reference#provides. Further one could argue, the packages from this repository do not provide compatibility, they do provide glibc.

`libc6-compat` is a package, https://pkgs.alpinelinux.org/packages?name=libc6-compat&branch=v3.16&repo=&arch=&maintainer=.

Reply:
@jalberti I think I see why setting `provide=libc6-compat` is not what we would want; after the suggested reading by @sgerrand :

>provides
>
> List of package names (and optionally version info) this package provides.
If package with a version is provided (provides='foo=1.2') apk will consider it as an **alternate name** and it will automatically consider the package for installation by the alternate name, and conflict with other packages having the same name, or provides.
>
> If version is not provided (provides='foo'), apk will consider it as virtual package name. Several package with same non-versioned provides can be installed simultaneously. However, none of them will be installed by default when requested by the virtual name - instead, error message is given and user is asked to choose which package providing the virtual name should be installed.

This definition shows that provides it meant as an alternate name for installation. Since this is glibc and not just a compatibility, it would make more since to set `provide='glibc=2.35'`

I'm not familiar with ABUILD, but there must be another way to show conflict between `libc6-compat` and other packages.

Reply:
@jalberti My this would work!?

depends="$pkgname bash !libc6-compat libgcc"

Notice the "!" added before libc6-compat. This makes the most since, and explains why that is in the depends list in the first place.

Excerpt from https://wiki.alpinelinux.org/wiki/APKBUILD_Reference#depends
>depends
>
>Run-time dependency package(s) that are not shared-object dependencies. Shared objects dependencies are auto-
detected and should not be specified here. To specify a conflicting package, add the package name prefixed with a '!'.